text/0000-docs-information-architecture

```diff
@@ -0,0 +1,154 @@
+        - dynamically rendered navigation*
+    - Deploying
+    - Hosting
+        - Path Prefix
+    - Performance
+        - Search Engine Optimization (SEO)
+        - Browser 
+        - Setting up proper HTTP caching
+- Migration Guides
+    - v0 to v1
+    - v1 to v2*
+    - v0 to v2 (good idea??)*
+- Gatsby components and libraries
+    - Glamor
+    - Styled components
+    - gatsby-image
+
+
+Plugins
+- Plugin Authoring Guides
+    - How Gatsby works with plugins
+    - Plugin library
+    - When to create your own plugin?*
+    - Source plugins
+    - Transformer plugins
+    - All other plugins
+    - How to get your plugin to show up in the Plugin Library*
+
+
+Contributing
+- Welcome to Gatsby's open source community!
+- Code of Conduct
+- How to Contribute
+    - List of ways to contribute*
+    - How to file an issue
+    - How to submit a PR*
+- Style Guides
+    - Written documents*
+    - Visual design*
+- Templates & Resources*
+    - Workshops
+    - PowerPoint presentations
+    - Blogpost
+    - Tutorial
+    - Plugin READme
+    - Site READme
+
+
+
+What we may not need to include in the docs:
+- Starters and the Awesome Gatsby stuff will appear in a starter library and a site showcase, so will no longer need to be in docs fairly soon.
+- Integration tutorials and guides (such as Gatsby + Netlify, Gatsby + Contentful). 
+    - Reasons to exclude these from docs: imho, we don't have the time to maintain the potentially infinite number of tutorials this could include. Also, perhaps we don't want to show preference for any company over another? Do we just make sure all the plugin READme files are really high quality?
+    - Reasons to still include them in docs or somewhere else (like a blog): tons and tons of people would use them.
+
+# Drawbacks
+
+Why should we *not* do this? Please consider:
+
+- I'm not sure what the code complexity of changing the URL structure of pages includes
+- The side bar might be able to remain the same size or structure. Not sure.
+- Could this break links people have to current docs?
+
+
+# Alternatives
+
+I'm not aware of any other designs we've considered. Anyone? The potential benefits of doing this outweight the negatives, imho. Especially because docs are only going to keep growing. The current hierarchy doesn't allow for much more growth.
+
+# Adoption strategy
+
+I don't think this would be hard for people to adopt, hopefully.
+
+# How we teach this
+
+If it's really good information architecture, it won't need to be taught ;). We would redesign it if we find that we have to explain it to people. We could write an initial blogpost with a tour of the docs structure for those who want to understand the change and see an overview, and then this overview could be included in an intro to the docs.
+
+# Unresolved questions
+
+- Number of levels of hierarchy. I want to keep it to 3 at the most. Suggestions or reasons to only do 2, or to do more, like 4?
+- How to handle change in all the URLs
```

Comment:
We'll need to make a list of old urls and new urls, then the site can be configured to send people from old to new. e.g.
`/docs/add-404-page/` changes to `/docs/basic-tasks/adding-content/add-404-page`